e2e/cypress/tests/integration/performance/utils.js

```diff
@@ -23,7 +23,8 @@ export const measurePerformance = (name, upperLimit, callback, reset, runs = 5)
 
         cy.log('loop finish').then(() => {
             const avgDuration = Math.round(durations.reduce((a, b) => a + b) / runs);
-            cy.log(`[PERFORMANCE] ${name}: ${avgDuration}ms` + avgDuration <= upperLimit ? ' within upper limit' : ' outside upper limit', ` of ${upperLimit}ms`);
+            cy.log(`[PERFORMANCE] ${name}: ${avgDuration}ms ${avgDuration <= upperLimit ? ' within upper limit' : ' outside upper limit'} of ${upperLimit}ms`);
+            cy.writeFile(`./tests/integration/performance/logs/${name}_${Date.now()}.json`, {name, date: Date.now(), duration: avgDuration});
```

Comment:
Minor nit, but could you put the timestamp into a variable before this line so that we're only calling `Date.now()`? I think the chance of this actually happening is pretty low, but it is possible that these timestamps could be different with how this is written currently